Cannot use `sys.stdout` in `create_subprocess_exec`.

* **uvloop version**: 0.9.1
* **Python version**: 3.6.5
* **Platform**: macOS 10.13.3
* **Can you reproduce the bug with `PYTHONASYNCIODEBUG` in env?**: Same behavior irregardless of `PYTHONASYNCIODEBUG`.

Code for reproduction:

```python
import asyncio
import sys

import uvloop


async def run_subprocess():
    proc = await asyncio.create_subprocess_exec(
        sys.executable, '-c', 'print(1)',
        stdout=sys.stdout,
    )
    await proc.communicate()


def run(loop_factory):
    loop = loop_factory()
    asyncio.set_event_loop(loop)
    try:
        loop.run_until_complete(run_subprocess())
    finally:
        loop.close()


def main():
    print('asyncio')
    run(asyncio.new_event_loop)
    print('uvloop')
    run(uvloop.new_event_loop)


if __name__ == '__main__':
    main()
```

Output:

```python-traceback
asyncio
1
uvloop
Traceback (most recent call last):
  File "hoge.py", line 32, in <module>
    main()
  File "hoge.py", line 28, in main
    run(uvloop.new_event_loop)
  File "hoge.py", line 19, in run
    loop.run_until_complete(run_subprocess())
  File "uvloop/loop.pyx", line 1364, in uvloop.loop.Loop.run_until_complete
  File "hoge.py", line 10, in run_subprocess
    stdout=sys.stdout,
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/coroutines.py", line 110, in __next__
    return self.gen.send(None)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/subprocess.py", line 225, in create_subprocess_exec
    stderr=stderr, **kwds)
  File "uvloop/loop.pyx", line 2368, in __subprocess_run
  File "uvloop/handles/process.pyx", line 564, in uvloop.loop.UVProcessTransport.new
  File "uvloop/handles/process.pyx", line 49, in uvloop.loop.UVProcess._init
  File "uvloop/handles/process.pyx", line 38, in uvloop.loop.UVProcess._init
  File "uvloop/handles/process.pyx", line 254, in uvloop.loop.UVProcess._init_options
  File "uvloop/handles/process.pyx", line 456, in uvloop.loop.UVProcessTransport._init_files
ValueError: invalid stdout argument value 1
```

@1st1 

```
    process = await asyncio.create_subprocess_exec(*cmd)
../../../../../.pyenv/versions/3.7.8/lib/python3.7/asyncio/subprocess.py:217: in create_subprocess_exec
    stderr=stderr, **kwds)
uvloop/loop.pyx:2749: in subprocess_exec
    ???
uvloop/loop.pyx:2707: in __subprocess_run
    ???
uvloop/handles/process.pyx:596: in uvloop.loop.UVProcessTransport.new
    ???
uvloop/handles/process.pyx:60: in uvloop.loop.UVProcess._init
    ???
uvloop/handles/process.pyx:49: in uvloop.loop.UVProcess._init
    ???
uvloop/handles/process.pyx:274: in uvloop.loop.UVProcess._init_options
    ???
uvloop/handles/process.pyx:458: in uvloop.loop.UVProcessTransport._init_files
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_pytest.capture.DontReadFromInput object at 0x114fc6590>

    def fileno(self) -> int:
>       raise UnsupportedOperation("redirected stdin is pseudofile, has no fileno()")
E       io.UnsupportedOperation: redirected stdin is pseudofile, has no fileno()

../../.venv/lib/python3.7/site-packages/_pytest/capture.py:230: UnsupportedOperation
```

--- a/tests/test_process.py
+++ b/tests/test_process.py
@@ -15,6 +15,20 @@ import psutil
 from uvloop import _testbase as tb
 
 
+class _RedirectFD(contextlib.AbstractContextManager):
+    def __init__(self, old_file, new_file):
+        self._old_fd = old_file.fileno()
+        self._old_fd_save = os.dup(self._old_fd)
+        self._new_fd = new_file.fileno()
+
+    def __enter__(self):
+        os.dup2(self._new_fd, self._old_fd)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        os.dup2(self._old_fd_save, self._old_fd)
+        os.close(self._old_fd_save)
+
+
 class _TestProcess:
     def get_num_fds(self):
         return psutil.Process(os.getpid()).num_fds()
@@ -407,6 +421,36 @@ print("OK")
 
         self.loop.run_until_complete(main())
 
+    def test_process_streams_redirect(self):
+        async def test():
+            prog = bR'''
+import sys
+print('out', flush=True)
+print('err', file=sys.stderr, flush=True)
+            '''
+
+            proc = await asyncio.create_subprocess_exec(
+                sys.executable, b'-W', b'ignore', b'-c', prog)
+
+            out, err = await proc.communicate()
+            self.assertIsNone(out)
+            self.assertIsNone(err)
+
+        with tempfile.NamedTemporaryFile('w') as stdout:
+            with tempfile.NamedTemporaryFile('w') as stderr:
+                with _RedirectFD(sys.stdout, stdout):
+                    with _RedirectFD(sys.stderr, stderr):
+                        self.loop.run_until_complete(test())
+
+                stdout.flush()
+                stderr.flush()
+
+                with open(stdout.name, 'rb') as so:
+                    self.assertEqual(so.read(), b'out\n')
+
+                with open(stderr.name, 'rb') as se:
+                    self.assertEqual(se.read(), b'err\n')
+
 
 class _AsyncioTests:
 
@@ -752,38 +796,7 @@ print(n)'''
 
 
 class Test_UV_Process(_TestProcess, tb.UVTestCase):
-
-    def test_process_streams_redirect(self):
-        # This won't work for asyncio implementation of subprocess
-
-        async def test():
-            prog = bR'''
-import sys
-print('out', flush=True)
-print('err', file=sys.stderr, flush=True)
-            '''
-
-            proc = await asyncio.create_subprocess_exec(
-                sys.executable, b'-W', b'ignore', b'-c', prog)
-
-            out, err = await proc.communicate()
-            self.assertIsNone(out)
-            self.assertIsNone(err)
-
-        with tempfile.NamedTemporaryFile('w') as stdout:
-            with tempfile.NamedTemporaryFile('w') as stderr:
-                with contextlib.redirect_stdout(stdout):
-                    with contextlib.redirect_stderr(stderr):
-                        self.loop.run_until_complete(test())
-
-                stdout.flush()
-                stderr.flush()
-
-                with open(stdout.name, 'rb') as so:
-                    self.assertEqual(so.read(), b'out\n')
-
-                with open(stderr.name, 'rb') as se:
-                    self.assertEqual(se.read(), b'err\n')
+    pass
 
 
 class Test_AIO_Process(_TestProcess, tb.AIOTestCase):

--- a/uvloop/handles/process.pyx
+++ b/uvloop/handles/process.pyx
@@ -452,7 +452,7 @@ cdef class UVProcessTransport(UVProcess):
             else:
                 io[0] = self._file_redirect_stdio(_stdin)
         else:
-            io[0] = self._file_redirect_stdio(sys.stdin.fileno())
+            io[0] = self._file_redirect_stdio(0)
 
         if _stdout is not None:
             if _stdout == subprocess_PIPE:
@@ -480,7 +480,7 @@ cdef class UVProcessTransport(UVProcess):
             else:
                 io[1] = self._file_redirect_stdio(_stdout)
         else:
-            io[1] = self._file_redirect_stdio(sys.stdout.fileno())
+            io[1] = self._file_redirect_stdio(1)
 
         if _stderr is not None:
             if _stderr == subprocess_PIPE:
@@ -508,7 +508,7 @@ cdef class UVProcessTransport(UVProcess):
             else:
                 io[2] = self._file_redirect_stdio(_stderr)
         else:
-            io[2] = self._file_redirect_stdio(sys.stderr.fileno())
+            io[2] = self._file_redirect_stdio(2)
 
         assert len(io) == 3
         for idx in range(3):

